core/src/main/scala/kafka/utils/FileLock.scala

```diff
@@ -65,8 +65,14 @@ class FileLock(val file: File) extends Logging {
   def unlock(): Unit = {
     this synchronized {
       trace(s"Releasing lock on ${file.getAbsolutePath}")
-      if (flock != null)
+      if (flock != null) {
         flock.release()
+        if (file.delete()) {
+          trace(s"Deleted ${file.getAbsolutePath}")
+        } else {
+          warn(s"Could not delete ${file.getAbsolutePath}")
+        }
```

Comment:
IMO, it is confusing if we do file deletion in `unlock` method. Suppose we need unlock and then lock, it'll fail now, right? Maybe delete in `destroy`?

Reply:
Agree! Deleting the file in `destroy` makes sense to me. I will update it.